### PR TITLE
CI: Use Playwright 1.17

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR updates the version of Playwright used to 1.17.

Due to issues with the version of Chromium contained in Playwright 1.17, this PR also forces the usage of the `Chrome` channel. [Playwright 1.17 comes bundled with Chrome 96](https://github.com/microsoft/playwright/releases/tag/v1.17.0).

Use of a more up-to-date version of Playwright has the following benefits:
- APIs introduced since 1.14
- Trace file supports rich page previews again.
- test against the current version of Chrome, the dominant browser used by our customers.

### CODE REVIEW

File: packages/calypso-e2e/src/browser-manager.ts [hunk 1]
```diff
@@ -84,6 +84,7 @@ export async function startBrowser( browserType: BrowserType ): Promise< Browser
 	}
 
 	browser = await browserType.launch( {
+		channel: 'chrome',
```
Comment (worldomonation) [inline packages/calypso-e2e/src/browser-manager.ts]: I wanted to avoid this change, but it appears this is necessary if we are to move off Playwright 1.14.

In TeamCity, using the bundled Chromium browser on Playwright version 1.14+ leads to either:

Outright failure to run any tests 
image

or

Handful of tests failing persistently
image

It is not known for sure why this is the case, as running this locally with Chromium (headless and headed) does not reproduce the issue. Even more oddly, these failures do not result in trace files or screenshots/videos being uploaded as artifact, making debugging quite difficult.

---

It is also possible to use the bundled Chromium version to obtain a passing run, but that would mean running this in headed mode and using `xvfb-run`. Performance impact appears minimal from what I've seen, though.

Comment (worldomonation) [reply packages/calypso-e2e/src/browser-manager.ts]: Another thought occurred to me: could this possibly due to our custom docker image? 

For reference, [this is the docker image](https://github.com/microsoft/playwright/blob/master/utils/docker/Dockerfile.bionic) supplied by the Playwright team.

Comment (scinos) [reply packages/calypso-e2e/src/browser-manager.ts]: I tested it locally using our latest docker image + Xquartz and running `yarn jest ./specs/specs-playwright/wp-gutenboarding__happy-path.ts` and works for me both headless and headed mode

**Comment by scinos** [comment]
I working with the theory that the build where a bunch of tests fails is correct and those tests have some kind of problem in Chromium and/or Playwright 1.17. Whit that theory, I'm focusing on fixing why there are no screenshots/traces for those tests.

So far I've found that the problem is `page.screenshot()`. I can see how it is being called both in our logs and in PW debug logs, but it times out after 30s with no info. I don't think it is related to the docker image because it does take the screenshot when running the image locally. Maybe permissions when it's run in TeamCity?

I'll continue to update this comment with my findings.

**Comment by worldomonation** [comment]
> I working with the theory that the build where a bunch of tests fails is correct and those tests have some kind of problem in Chromium and/or Playwright 1.17. Whit that theory, I'm focusing on fixing why there are no screenshots/traces for those tests.

What you describe is a possibility - Playwright 1.17 changed element targeting behavior has been changed (see https://github.com/microsoft/playwright/pull/9546). Other fixes between 1.14 and 1.17 could have removed a behavior that I relied on to model interaction with the page.

On the other hand - Chromium version seemed like the first culprit, but recall that I also ran into similar issues to this PR in a previous iteration (https://github.com/Automattic/wp-calypso/pull/57773). Chromium versions contained in each Playwright release runs quite far ahead of Chrome, but eventually Chrome would catch up. 

While Chromium !== Chrome, I'd have expected Chrome 94 in Playwright 1.16 to surface the same problems as Chromium 96 in Playwright 1.15.2. However this does not happen.

---

To recap the current situation:
| Playwright version | Chromium version | Chrome version |
| --- | --- | ---  |
| 1.14 |  94.0.4595.0  | 92  |
| 1.15.2  | 96.0.4641.0 | 93 |
| 1.16  |  97.0.4666.0 |  94 |
 | 1.17  | 98.0.4695.0 | 96 |

In the local environment, all of these versions work fine - headful and headless.
In CI - 
- Chromium versions after 1.14 leads to unexpected failures when headless.
- Chrome functions as expected, headful or headless.

**Comment by scinos** [comment]
Why do we want to run tests in Chromium instead of Chrome?

**Comment by scinos** [comment]
More about the screenshot problem. TLDR: I still don't know what is going on.

These the relevant traces (I think) when it "works" (as in, I run `channel: "chrome"`, make a test fail and can take a screen shot):

```
2021-12-03T17:35:20.364Z pw:protocol SEND ► {"id":130,"method":"Page.captureScreenshot","params":{"format":"png","clip":{"x":0,"y":0,"width":1280,"height":720,"scale":1}},"sessionId":"48D8005E382F9851129CA43E6A5978AD"}
... bunch of traffic here, loading images, scripts, etc. ...
2021-12-03T17:35:20.728Z pw:protocol ◀ RECV {"id":130,"result":{"data":"iVBORw0KGgoAAAANSUhEUgAACgAAAAWgCAYAAACSABrfAAAAAXNSR0IAr...
```

However, when it fails (i.e using Chromium), I can see the `Page.captureScreenshot` SEND, but not the corresponding RECV. Interestingly, what I do see is a couple of network requests failing right after `Page.captureScreenshot`. But I don't know if that's related or just noise.

Build where it fails: https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/7074855?showLog=7074855_7391_325.7441.7442.7443.7446
Build where it works (as in, it can take the screenshot): https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/7076199?showLog=7076199_6785_321.7036.7037.7038.7041

----

Update:

Interestingly, traces seem to work (https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/7077394?buildTab=artifacts#%2Ftrace;%2Flogs.tgz). Screenshots timeout and video records fail with:

```
21:47:58       Trace: Error: Page did not produce any video frames
21:47:58           at Video.saveAs (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/node_modules/playwright-core/lib/client/video.js:47:26)
21:47:58           at /home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/hooks.ts:98:5
21:47:58           at /home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/hooks.ts:101:13
```

---
Update:

Found this related issue https://github.com/microsoft/playwright/issues/10460.

**Comment by worldomonation** [comment]
> Why do we want to run tests in Chromium instead of Chrome?

There isn't a particular reason for preferring Chromium over Chrome in our use case.

The [rationale](https://playwright.dev/docs/browsers#chromium) cited by the Playwright development team in favor of Chromium is around testing ahead of the currently release version in order to detect and fix any compatibility and regression issues.

For our purposes though, I don't see a reason why we **need** to stick with Chromium. In fact, Chrome is better representative of what our end users would likely encounter.

We are currently around two versions behind (Chromium 94 vs Chrome 96). My opinion is that perhaps it's time to switch to Chrome for regular testing aside from the ongoing investigation into why the failures occur in `screenshot`.

**Comment by scinos** [comment]
Update:

Playwright 1.16.3 seems to work fine (as in, it takes screenshots when the tests fails)

**Comment by scinos** [comment]
Update:

Playwright 1.16.3 seems to work fine (as in, it takes screenshots when the tests fail)

**Comment by scinos** [comment]
After looking at this for a few hours, my understanding is:

- Playwright 1.17 introduces some changes that can make our tests flakey. That's ok (for the purpose of this comment).
- Playwright 1.17 + Chromium can't take screenshots or video recordings when a test fails.
- Playwright 1.17 + Chrome works fine (it takes screenshots and videos).
- Playwright 1.16.3 + Chromium works fine (it takes screenshots and videos).
- There is nothing in Playwright logs (not even with DEBUG=*) that helps diagnose this problem.

While it may be related to running inside a Docker container, my guess is that this is an actual bug inside Playwright, specifically with the Chromium version of the bundle. https://github.com/microsoft/playwright/issues/10460 suggests the same.

I'd say we either downgrade to 1.16.3 until it is resolved or switch the Chrome

**Comment by worldomonation** [comment]
> After looking at this for a few hours, my understanding is:
> 
>     * Playwright 1.17 introduces some changes that can make our tests flakey. That's ok (for the purpose of this comment).
> 
>     * Playwright 1.17 + Chromium can't take screenshots or video recordings when a test fails.
> 
>     * Playwright 1.17 + Chrome works fine (it takes screenshots and videos).
> 
>     * Playwright 1.16.3 + Chromium works fine (it takes screenshots and videos).
> 
>     * There is nothing in Playwright logs (not even with DEBUG=*) that helps diagnose this problem.
> 
> 
> While it may be related to running inside a Docker container, my guess is that this is an actual bug inside Playwright, specifically with the Chromium version of the bundle. [microsoft/playwright#10460](https://github.com/microsoft/playwright/issues/10460) suggests the same.
> 
> I'd say we either downgrade to 1.16.3 until it is resolved or switch the Chrome

Interesting find. I pushed a branch with Playwright 1.16.3 pinned but was not able to achieve similar results as you did with 1.16.3. 

The part that makes me slightly suspicious is that even using the same version of Chromium the behavior is altered depending on `headful` or `headless`.

In general, my opinion is:
- there is little value in using bleeding edge version of Chromium
- there is value to be gained by using latest version of Playwright.

**Comment by scinos** [comment]
> In general, my opinion is:
> * there is little value in using bleeding edge version of Chromium
> * there is value to be gained by using latest version of Playwright.

That sounds very reasonable to me

**Review by scinos — APPROVED** [review]
